src/main/java/org/pesmypetcare/webservice/dao/MedicationDao.java

```diff
@@ -0,0 +1,93 @@
+
+package org.pesmypetcare.webservice.dao;
+
+import org.pesmypetcare.webservice.entity.MedicationEntity;
+import org.pesmypetcare.webservice.error.DatabaseAccessException;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+
+public interface MedicationDao {
+    /**
+     * Creates a medication eaten by a pet on the database.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param dateName Date + name of the receipt of the medication
+     * @param medication The medication entity that has the attributes of the medication for the pet.
+     */
+    void createMedication(String owner, String petName, String dateName, MedicationEntity medication);
+
+    /**
+     * Deletes all the medication with the specified owner and pet from certain date.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param dateName Date + name of the medication to delete
+     */
+    void deleteByDateAndName(String owner, String petName, String dateName);
+
+
+    /**
+     * Deletes all the medications of the specified pet from database.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    void deleteAllMedications(String owner, String petName) throws DatabaseAccessException;
+
+    /**
+     * Gets a medication identified by its pet, date and name
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param dateName pk of the medication
+     * @return The MedicationEntity identified by the data
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    MedicationEntity getMedicationData(String owner, String petName, String dateName) throws DatabaseAccessException;
+
+    /**
+     * Gets the data from all the specified medications from the database identified by its pet.
+     * @param owner Username of the owner of the pets
+     * @param petName Name of the pet
+     * @return The List containing all the medications of the pet
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    List<Map<List<String>, Object>> getAllMedicationData(String owner, String petName) throws DatabaseAccessException, ExecutionException, InterruptedException;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 160).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)